service/http.balancer/src/test/java/org/kaazing/gateway/service/http/balancer/SecureBalancerIT.java

```diff
@@ -0,0 +1,151 @@
+
+public class SecureBalancerIT {
+
+    private static KeyStore keyStore;
+    private static KeyStore trustStore;
+    private static char[] password;
+
+    private final K3poRule k3po = new K3poRule();
+    private final TestRule timeout = new DisableOnDebug(new Timeout(10, SECONDS));
+    private final MethodExecutionTrace trace = new MethodExecutionTrace();
+
+    private final GatewayRule balancerGateway = new GatewayRule() {
+        {
+            GatewayConfiguration configuration = new GatewayConfigurationBuilder()
+                    .service()
+                        .type("balancer")
+                        .accept("wss://localhost:8002/echo")
+                        .crossOrigin()
+                            .allowOrigin("*")
+                        .done()
+                    .done()
+                    .service()
+                        .type("echo")
+                        .accept("wss://localhost:8003/echo")
+                        .balance("wss://localhost:8002/echo")
+                        .crossOrigin()
+                            .allowOrigin("*")
+                        .done()
+                    .done()
+                    .security()
+                        .keyStore(keyStore)
+                        .keyStorePassword(password)
+                    .done()
+            .done();
+
+            init(configuration);
+        }
+    };
+
+    private final GatewayRule proxyGateway = new GatewayRule() {
+        {
+            GatewayConfiguration configuration = new GatewayConfigurationBuilder()
+                    .service()
+                        .type("proxy")
+                        .accept("tcp://localhost:8001/")
+                        .connect("ssl://localhost:8002")
+                        .crossOrigin()
+                            .allowOrigin("*")
+                        .done()
+                    .done()
+                    .security()
+                        .trustStore(trustStore)
+                    .done()
+            .done();
+
+            init(configuration);
+        }
+    };
+
+
+    @Rule
+    public TestRule chain = outerRule(trace).around(balancerGateway).around(proxyGateway).around(k3po).around(timeout);
+
+    @BeforeClass
+    public static void initClass() throws Exception {
+        password = "ab987c".toCharArray();
+
+        /// Initialize gateway keystore
+        keyStore = KeyStore.getInstance("JCEKS");
+        FileInputStream kis = new FileInputStream("target/truststore/keystore.db");
+        keyStore.load(kis, password);
+        kis.close();
+
+        // Initialize gateway truststore 
+        trustStore = KeyStore.getInstance("JKS");
+        FileInputStream tis = new FileInputStream("target/truststore/truststore.db");
+        trustStore.load(tis, null);
+        tis.close();
+    }
+
+    @Test
+    @Specification("wsx.balancer.request")
+    @ScriptProperty({"hostHeader 'localhost:8002'", "redirectAddress 'https://localhost:8003/echo'"})
```

Comment:
It has always been that way. The scripts were already written and used in the non-secure tests, I just figured I could reuse them for the secure ones. So, I just updated them to use properties, so that I could change the expected value of the location header.